src/local.js

```diff
@@ -0,0 +1,117 @@
+const assert = require('assert')
+const composer = require('./composer')
+const docker = require('./docker')
+const fs = require('fs').promises
+const path = require('path')
+const npm = require('./npm')
+const { fileNamesOfInterest, assertFileNameIsInListWeSearchFor } = require('./file')
+
+const foldersToExclude = [
+  '.git',
+  'vendor',
+  'node_modules'
+]
+
+/**
+ * Get the base images used in each of the specified repo's Dockerfiles.
+ *
+ * @returns {Promise<string[]>}
+ */
+const getDockerBaseImagesOfRepo = async (directory) => {
+  const contentsOfFiles = await getContentsOfFilesByName(directory, 'Dockerfile')
+  const allBaseImages = []
+  for (const contentsOfFile of contentsOfFiles) {
+    const fileBaseImages = docker.getBaseImagesFrom(contentsOfFile)
+    allBaseImages.push(...fileBaseImages)
+  }
+  return allBaseImages
+}
+
+/**
+ * Get the JavaScript dependencies of the specified repo.
+ *
+ * @returns {Promise<undefined | {name: string, version: string}[]>}
+ */
+const getJavaScriptDependenciesOfRepo = async (directory) => {
+  const contents = await getContentsOfFileByName(directory, 'package-lock.json')
+  return npm.getDependenciesFromPackageLock(contents)
+}
+
+/**
+ * Get the PHP dependencies of the specified repo.
+ *
+ * @returns {Promise<undefined | {name: string, version: string}[]>}
+ */
+const getPhpDependenciesOfRepo = async (directory) => {
+  const contents = await getContentsOfFileByName(directory, 'composer.lock')
+  return composer.getDependenciesFromComposerLock(contents)
+}
+
+const getContentsOfFileByName = async (directory, fileName) => {
+  const file = await findFile(directory, fileName)
+  let contents = undefined
+  try {
+    const buff = await fs.readFile(file)
+    contents = buff.toString()
+  } catch (err) {
+    console.error(`Error (${err.status}): ${err.message}`)
+  }
+  return contents
+}
+
+const getContentsOfFilesByName = async (directory, fileName) => {
+  const files = await findFiles(directory, fileName)
+  return await files.map(async file => {
+    try {
+      const buff = await fs.readFile(file)
+      return buff.toString()
+    } catch (err) {
+      console.error(`Error (${err.status}): ${err.message}`)
```

Comment:
I don't know whether file-system related errors will have a `.status` property. You may need to tweak this error message a bit, something like...
```suggestion
      console.error(`Error reading "${file}": ${err.message}`)
```